L1Trigger/L1THGCal/src/HGCalTriggerTools.cc

```diff
@@ -184,8 +193,15 @@ float HGCalTriggerTools::getLayerZ(const int& subdet, const unsigned& layer) con
   } else if(subdet == ForwardSubdetector::HGCHEF) {
     layerGlobalZ = geom_->fhTopology().dddConstants().waferZ(layer, true);
   } else if(subdet == HcalSubdetector::HcalEndcap || subdet == ForwardSubdetector::HGCHEB) {
-    std::pair<int,int> eta_range = geom_->bhTopology().dddConstants()->getEtaRange(1);
-    layerGlobalZ = geom_->bhTopology().dddConstants()->getRZ(HcalSubdetector::HcalEndcap,eta_range.second,layer);
+    if(geom_->bhGeometry())
+    {
+      std::pair<int,int> eta_range = geom_->bhTopology().dddConstants()->getEtaRange(1);
```

Comment:
could avoid the temporary here by putting `geom_->bhTopology().dddConstants()->getEtaRange(1).second` into the next line